openapi.yaml

```diff
@@ -1,279 +1,662 @@
-    maximum: 500
-    minimum: 1
-responses:
-  '400':
-     description: Bad request
-     schema:
-       $ref: '#/definitions/ErrorResult'
-  '500':
-     description: Internal server error
-     schema:
-       $ref: '#/definitions/ErrorResult'
-securityDefinitions:
-  OAuth2:
-    type: oauth2
-    flow: application
-    tokenUrl: https://api.oregonstate.edu/oauth2/token
-    scopes:
-      full: Full access to the API
-definitions:
-  Meta:
-    properties:
-      totalResults:
-        type: integer
-        description: Total number of results
-        example: 10
-      totalPages:
-        type: integer
-        description: Total number of pages
-        example: 10
-      currentPageNumber:
-        type: integer
-        description: Page number of the returned results
-        example: 1
-      currentPageSize:
-        type: integer
-        description: Number of results per page
-        example: 25
-  SelfLink:
-    properties:
-      self:
-        type: string
-        format: uri
-        description: Self-link of current resource
-  PaginationLinks:
-    properties:
-      first:
-        type: string
-        format: uri
-        description: The first page of data
-      last:
-        type: string
-        format: uri
-        description: The last page of data
-      prev:
-        type: string
-        format: uri
-        description: The previous page of data
-      next:
-        type: string
-        format: uri
-        description: The next page of data
-  PetResource:
-    properties:
-      id:
-        type: string
-        description: A unique ID of pet
-      type:
-        $ref: '#/definitions/PetType'
-      links:
-        $ref: '#/definitions/SelfLink'
-      attributes:
-        properties:
-          name:
             type: string
-            description: Name of pet
-            example: Hedwig
-          owner:
+      responses:
+        "200":
+          description: "Successful response"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/ServicesResultObject"
+        "404":
+          description: "Resource not found"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Errors"
+        "500":
+          description: "Internal Server Error"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Errors"
+components:
+  parameters:
+    pretty:
+      name: pretty
+      in: query
+      required: false
+      description: If true, JSON response will be pretty-printed
+      schema:
+        type: boolean
+  securitySchemes:
+    OAuth2:
+      type: oauth2
+      flows: 
+        clientCredentials:
+          tokenUrl: https://api.oregonstate.edu/oauth2/token
+          refreshUrl: https://api.oregonstate.edu/oauth2/token
+          scopes: 
+            full: full access to the API
```

Comment:
Not positive this is the correct way to implement this, but it's my best try based on the [available documentation](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#oauth-flows-object)

Reply:
Looks good to me

Reply:
You can just use what we have defined in the skeleton: https://github.com/osu-mist/express-api-skeleton/blob/2e334d2061be607ddb09e417cbdfe4d5febcc81a/openapi.yaml#L121-L127

Reply:
Thanks, I didn't realize we already had an openapi v3 doc for the skeleton